tests/browser/async_virtual_2.cpp

```diff
@@ -16,7 +16,7 @@ class DOS_Device {
 DOS_Device *Devices[10];
 
 bool __attribute__((noinline)) DOS_Device::Read(unsigned char * data,unsigned short * size) {
-    printf("DOS_Device::Read (this = %i)\n", (int)this);
+    printf("DOS_Device::Read (this = %ld)\n", (long)this);
```

Comment:
This should just be %p

Reply:
Yup, that be better.. I was just trying to fix the existing style which seems to like to cast to `int`.
411 results for `%ld` (most of which "pointers", only 115 of which from PR) - might want to leave this improvement for another PR though.

Reply:
Maybe lets land that first?  That would make this PR a lot smaller.   I can have a go at it if you like?

Reply:
Wouldn't that produce a ton of conflicts with this PR?
I don't think we should mix trying to make things 64-bit compatible with general large scale cleanup of tests. I am sure if you go look at the tests you can find a lot of things to clean up.

Reply:
I think its good to do such cleanups when it becomes apparent that they are useful/needed.   In this case doing the cleanup ahead of time will significantly reduce the size of, and churn generated by, this PR. 

Yes it will conflict with this PR but it should be trivial to just run `git checkout --thiers` to accept the upstream changes since we know we want to basically revert all the conflicting parts of this PR, right?

Reply:
(i.e. there should be no manual conflict resolution needed I think)